File: webui/src/widgetCaps.ts
/** 与 internal/layout/widget_caps.go 对齐：多维度 / 多行表达式能力 */

export function widgetAllowsMultiDims(widgetType: string): boolean {
  return widgetType === 'histogram';
}

export function widgetAllowsMultiExpr(widgetType: string): boolean {
  return widgetType === 'histogram';
}

File: internal/layout/widget_caps.go
package layout

// WidgetAllowsMultiDims 是否允许 dimensions 中多个维度（英文逗号分隔）。
func WidgetAllowsMultiDims(t WidgetType) bool {
	return t == WidgetHistogram
}

// WidgetAllowsMultiExpr 是否允许 value_expr 多行（复合模式下多条表达式）。
func WidgetAllowsMultiExpr(t WidgetType) bool {
	return t == WidgetHistogram
}
